src/charm.py

```diff
@@ -143,6 +148,44 @@ def snap_path(self) -> Optional[str]:
 
         return self._snap_path
 
+    @property
+    def snap_channel(self) -> str:
+        """Get the channel for exporter snap.
+
+        The channel is determined by the controller version that the charm is deployed
+        under. In case the controller version is equal or higher than 2.6 and less
+        than 2.9, the snap channel is set to 2.8/stable. In case controller's major and
+        minor version match with 2.9, the snap channel is set to 2.9/stable. Otherwise,
+        raise ControllerIncompatibleError exception.
+        """
+        controller_version = self.get_controller_version()
+
+        if controller_version < version.parse("2.6"):
```

Comment:
using `elif` after `return` or `raise` will be flagged by pylint because it is unnecessary. And since I'm making "less-than", "large-than" comparison rather than exact matching, `if` would be the best fit here IMO.